zero_bin/common/src/prover_state/persistence.rs

```diff
@@ -17,10 +19,20 @@ use super::{
     Config, RecursiveCircuitsForTableSize, SIZE,
 };
 
-const CIRCUITS_DIR: &str = "circuits/";
 const PROVER_STATE_FILE_PREFIX: &str = "prover_state";
 const VERIFIER_STATE_FILE_PREFIX: &str = "verifier_state";
-const CARGO_WORKSPACE_DIR_ENV: &str = "CARGO_WORKSPACE_DIR";
+const ZK_EVM_CACHE_DIR_NAME: &str = "zk_evm_circuit_cache";
+const ZK_EVM_CACHE_DIR_ENV: &str = "ZK_EVM_CACHE_DIR";
+
+static CIRCUIT_CACHE_DIR: Lazy<String> = Lazy::new(|| {
```

Comment:
Not a huge fan of this - Is there any blocker to having the user pass it in as an argument when required?

```rust
struct Config {
    cache_dir: PathBuf,
}
impl Default for Config { .. }
fn entrypoint(config: impl Into<Option<Config>>, ...) { .. }
```

It would also make me very happy to stick these all in a `struct CacheProvider` somewhere, or find a library that abstracts caching for us

Reply:
So we could simplify this and just make it a `&'static str` constant instead. This is only accessed once in the binary, so I agree that making this lazy might be a bit overkill in hindsight.

Right now we sort of have this behavior with the `env` variable where it's not set if the user has already set it. We could move away from the `env` variable and solely to an optional prog arg (sorry if I'm misunderstanding your comment).

Reply:
How opposed would we be if this remained an `env` var? Every way I tried passing this down as a prog arg ends up being pretty terrible and messy.

Reply:
Fine by me